// responsive-react/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Responsive", {
  enumerable: true,
  get: function get() {
    return _Responsive.Responsive;
  }
});
Object.defineProperty(exports, "getWindowDimension", {
  enumerable: true,
  get: function get() {
    return _utilResponsive.getWindowDimension;
  }
});
Object.defineProperty(exports, "getDeviceTypeInfo", {
  enumerable: true,
  get: function get() {
    return _utilResponsive.getDeviceTypeInfo;
  }
});
Object.defineProperty(exports, "isMobileDevice", {
  enumerable: true,
  get: function get() {
    return _utilResponsive.isMobileDevice;
  }
});
Object.defineProperty(exports, "isTabletDevice", {
  enumerable: true,
  get: function get() {
    return _utilResponsive.isTabletDevice;
  }
});
Object.defineProperty(exports, "isLaptopDevice", {
  enumerable: true,
  get: function get() {
    return _utilResponsive.isLaptopDevice;
  }
});
Object.defineProperty(exports, "isBiggerThanLaptop", {
  enumerable: true,
  get: function get() {
    return _utilResponsive.isBiggerThanLaptop;
  }
});

var _Responsive = require("./Responsive");

var _utilResponsive = require("./utilResponsive");
